services/src/test/scala/cromwell/pipeline/service/RunServiceTest.scala

```diff
@@ -41,40 +33,47 @@ class RunServiceTest extends AsyncWordSpec with Matchers with MockitoSugar {
       runUpdated.cmwlWorkflowId
     )
 
-    "addRun" should {
+    def defaultRunRepository: RunRepositoryTestImpl = RunRepositoryTestImpl(run)
+    val defaultProjectService: ProjectServiceTestImpl = ProjectServiceTestImpl(project)
 
+    def createRunService(
+      runRepository: RunRepositoryTestImpl = defaultRunRepository,
+      projectService: ProjectServiceTestImpl = defaultProjectService
+    ): RunService =
+      RunService(runRepository, projectService)
+
+    def runService: RunService = createRunService()
+
+    "addRun" should {
       "returns run id" taggedAs Service in {
         val runCreateRequest = RunCreateRequest(projectVersion = run.projectVersion, results = run.results)
-        when(projectService.getUserProjectById(projectId, userId)).thenReturn(Future.successful(project))
-        when(runRepository.addRun(any[Run])).thenReturn(Future.successful(runId))
 
         runService.addRun(runCreateRequest, projectId, userId).map {
-          _ shouldBe runId
+          _ shouldBe a[RunId]
```

Comment:
Obviously it will be `a[RunId]`, that's what the type system will enforce even without tests. Why did you need to change this assertion?

Reply:
In that case `repository.addRun` returns a random run Id, generated inside `runService.addRun`.
Can't think of the decent way of how to test it with current repository testImpl.

Reply:
In this case, there are two ways to fix is.
1) Make your assertion more explicit 
```suggestion
          _ => succeed
```
2) Change the test order and modify this one. You should move tests for `getRunByIdAndUser` method on top and then add a test for `addRun` method. This test will add a run, then fetch it with `getRunByIdAndUser` and compare that this is the same run as we expected.

Reply:
Great ideas, thanks! Fixed